java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryProvider.java

```diff
@@ -49,6 +51,21 @@ public MapDictionaryProvider(Dictionary... dictionaries) {
       }
     }
 
+    /**
+     * Initialize the map structure from another provider, but with empty vectors.
+     *
+     * @param other the {@link DictionaryProvider} to copy the ids and fields from
+     * @param allocator allocator to create the empty vectors
+     */
+    public void copyStructureFrom(DictionaryProvider other, BufferAllocator allocator) {
```

Comment:
Ditto here, we should note that it's only used from JNI. (Can JPype invoke a protected or private method? I guess not, but it would be good to signal that this isn't meant to be a public function really.) We could mark it [`@VisibleForTesting`](https://github.com/apache/arrow/blob/01b42d51ba97cb12c17e5b001c65b38741ea83ba/java/memory/memory-core/src/main/java/org/apache/arrow/util/VisibleForTesting.java)